src/build.ml

```diff
@@ -162,10 +162,35 @@ let of_result_map ?targets res ~f =
 let memoize name t =
   Memo { name; t; state = Unevaluated }
 
+(* This is to force the rules to be loaded for directories without
+   files when depending on [(source_tree x)]. Otherwise, we wouldn't
+   clean up stale directories in directories that contain no file. *)
+let depend_on_dir_without_files =
+  let pred = Predicate.create ~id:(lazy (String "false")) ~f:(fun _ -> false) in
+  fun dir ->
+    Paths_glob (File_selector.create ~dir pred) >>^ ignore
```

Comment:
FYI, I've added `Build.ignore` to make this a bit safer. E.g:

```
... |> Build.ignore
```

Reply:
Thanks for the info, I updated the code